util/topsql/reporter/reporter.go

```diff
@@ -344,55 +542,101 @@ func addEvictedIntoSortedDataPoints(others *dataPoints, evict *dataPoints) *data
 		others.TimestampList = evict.TimestampList
 		others.CPUTimeMsList = evict.CPUTimeMsList
 		others.CPUTimeMsTotal = evict.CPUTimeMsTotal
+		others.StmtExecCountList = evict.StmtExecCountList
+		others.StmtKvExecCountList = evict.StmtKvExecCountList
+		others.StmtDurationSumNsList = evict.StmtDurationSumNsList
+		others.tsIndex = evict.tsIndex
 		return others
 	}
 	length := len(others.TimestampList) + len(evict.TimestampList)
 	timestampList := make([]uint64, 0, length)
 	cpuTimeMsList := make([]uint32, 0, length)
+	stmtExecCountList := make([]uint64, 0, length)
+	stmtKvExecCountList := make([]map[string]uint64, 0, length)
+	stmtDurationSumList := make([]uint64, 0, length)
 	i := 0
 	j := 0
 	for i < len(others.TimestampList) && j < len(evict.TimestampList) {
 		if others.TimestampList[i] == evict.TimestampList[j] {
 			timestampList = append(timestampList, others.TimestampList[i])
 			cpuTimeMsList = append(cpuTimeMsList, others.CPUTimeMsList[i]+evict.CPUTimeMsList[j])
+			stmtExecCountList = append(stmtExecCountList, others.StmtExecCountList[i]+evict.StmtExecCountList[j])
+			stmtKvExecCountList = append(stmtKvExecCountList, mergeKvExecCountMap(others.StmtKvExecCountList[i], evict.StmtKvExecCountList[j]))
+			stmtDurationSumList = append(stmtDurationSumList, others.StmtDurationSumNsList[i]+evict.StmtDurationSumNsList[j])
 			i++
 			j++
 		} else if others.TimestampList[i] < evict.TimestampList[j] {
 			timestampList = append(timestampList, others.TimestampList[i])
 			cpuTimeMsList = append(cpuTimeMsList, others.CPUTimeMsList[i])
+			stmtExecCountList = append(stmtExecCountList, others.StmtExecCountList[i])
```

Comment:
This report part is very horrible 😭 Too many place to modify when adding a single metric.. Would you like to discover some solutions to refine it later?